mailbox/tools/indexer/src/main/java/org/apache/mailbox/tools/indexer/ReIndexerPerformer.java

```diff
@@ -285,19 +285,19 @@ public ReIndexerPerformer(MailboxManager mailboxManager,
             .switchIfEmpty(Mono.just(Result.COMPLETED));
     }
 
-    private Mono<Task.Result> reIndex(Either<Failure, ReIndexingEntry> entryOrFailure, ReprocessingContext reprocessingContext) {
-        return entryOrFailure.map(this::index)
-            .fold(failure -> {
-                failure.recordFailure(reprocessingContext);
-                return Mono.just(Result.PARTIAL);
-            }, result -> result.map(either -> recordIndexingResult(either, reprocessingContext)));
+    private Mono<Task.Result> reIndex(Either<Failure, ReIndexingEntry> failureOrEntry, ReprocessingContext reprocessingContext) {
+        return toMono(failureOrEntry.map(this::index))
+            .map(failureOrIndexingResult -> failureOrIndexingResult.getOrElseGet(Either::left))
```

Comment:
We could extract this technical step into a "flatMapRight" method toabstract away complexity?